src/desktop.md

```diff
@@ -6,195 +6,346 @@ toc: true
-the Dart and Flutter extensions (also referred to as editor plugins).
+## Entitlements and the App Sandbox
 
-### VS Code: Create and run a new project with desktop support
-Creating a new project with desktop support is no different than
-[creating a new Flutter project][flutter starter app] for other platforms.
-Running a Flutter desktop app is no different than
-[running and debugging in VS Code] for other target platforms:
- 1. Select **macOS** as a target device from the VS Code status bar.
- 2. Click the **Debug > Start Without Debugging** or press <kbd>F5</kbd>.
+macOS builds are configured by default to be signed,
+and sandboxed with App Sandbox.
+This means that if you want to confer specific
+capabilities or services on your macOS app,
+such as the following:
 
-![Desktop VS Code](/images/desktop/desktop_vscode.gif){:width="70%"}
+* Accessing the internet
+* Capturing movies and images from the built-in camera 
+* Accessing files
 
-{{site.alert.note}}
-  Is `macOS` not listed as one of your available devices?
-  Make sure to first run `flutter create`
-  to create the `macos` directory in your project root.
+Then you must set up specific _entitlements_ in Xcode.
+The following section tells you how to do this.
+
+### Managing the sandbox
+
+Managing sandbox settings is done in the
+`macos/Runner/*.entitlements` files. When editing
+these files, you shouldn't remove the original
+`Runner-DebugProfile.entitlements` exceptions
+(incoming network connections and JIT),
+as they are necessary for the `debug` and `profile`
+modes to function correctly.
+
+If you are used to managing entitlement files through
+the Xcode capabilities UI, be aware that the capabilities
+editor updates only one of the two files or,
+in some cases, it creates a whole new entitlements
+file and switches the project to use it for all configurations;
+either scenario causes issues. The recommended approach is
+to edit the files directly. Unless you have a very specific
+reason, you should always make identical changes to both files.
+
+If you keep App Sandbox enabled, you need to manage entitlements
+for your application when you add certain plugins or other native
+functionality. For instance, using the `file_chooser` plugin
+requires adding either the
+`com.apple.security.files.user-selected.read-only` or
+`com.apple.security.files.user-selected.read-write` entitlement.
+Another common entitlement is `com.apple.security.network.client`,
+which you must add if you make any network requests.
+
+Using App Sandbox is required if you plan to distribute your
+application in the App Store.
+
+{{site.alert.secondary}}
+  **Important:** The `com.apple.security.network.server`
+  entitlement, which allows incoming network connections,
+  is enabled by default only for `debug` and `profile`
+  modes (to enable the Dart observatory). [PENDING: Do you still use observatory or do you use Dart DevTools? Do we need to mention observatory here?]
```

Comment:
As far as I know observatory is still the name of the service in the Dart VM that all the debug services talk to (e.g., https://dart.dev/tools/dart-vm#observatory-options). Dart DevTools seems to be a front-end package of tools (which presumably talks to the observatory service).

Reply:
If you don't want to use the term "observatory", the relevant language on the Android page is: "Add the android.permission.INTERNET permission if your application code needs Internet access. The standard template does not include this tag but allows Internet access during development to enable communication between Flutter tools and a running app."

I'm fine with something like "... is enabled by default only for `debug` and `profile` modes to enable communication between Flutter tools and a running app"